edx_rest_framework_extensions/auth/jwt/tests/test_authentication.py

```diff
@@ -180,13 +183,21 @@ def test_authenticate_csrf_protected(self, mock_set_custom_attribute):
         request = RequestFactory().post('/')
 
         request.META[USE_JWT_COOKIE_HEADER] = 'true'
+        # Set a sample JWT cookie. We mock the auth response but we still want
+        # to ensure that there is jwt set because there is other logic that
+        # checks for the jwt to be set before moving forward with CSRF checks.
+        request.COOKIES[jwt_cookie_name()] = 'foo'
 
         with mock.patch.object(JSONWebTokenAuthentication, 'authenticate', return_value=('mock-user', "mock-auth")):
-            with self.assertRaises(PermissionDenied) as context_manager:
-                JwtAuthentication().authenticate(request)
+            if get_setting(ENABLE_FORGIVING_JWT_COOKIES):
+                assert JwtAuthentication().authenticate(request) is None
```

Comment:
Although this is possibly good enough, once we commit to the custom attributes, it might be good to verify that we have captured the CSRF failure.

Reply:
Marking as resolved.